### PR TITLE
Update the top level template json file based on ZCL extensions

### PR DESCRIPTION
**Pull request by brdandu**
- Add the zcl extension files to part of its crc check.
- If the crc changes based on the inclusion of extensions then reload the top level template json file and turn its IN_SYNC to true and turn the old on to false
- These changes make sure that when a zcl extension file (eg json files in zcl section such as cluster-to-component.json) is updated then everything from the top level json files are loaded again
- JIRA: ZAPP-1092

### CODE REVIEW

**Review by paulr34 — APPROVED** [review]
looks good

File: src-electron/util/env.js [hunk 1]
```diff
@@ -73,14 +73,25 @@ export function builtinSilabsZclGeneralXmlFile() {
...
 }
 
+/**
+ * Used to retrieve cluster-to-component-dependencies.json by zcl reload test in zcl-loader.test.js
+ *
+ * @returns path to cluster-to-component-dependencies.json file used by zcl-loader.test.js
+ */
+export function builtinZclExtensionClusterToComponentFile() {
+  return locateProjectResource(
+    './test/gen-template/zigbee2/cluster-to-component-dependencies.json'
```
Comment (tecimovic) [inline src-electron/util/env.js:91]: env.js is not test file. But it refers to a test file here. You should put this in `test-util`, not env.js

Here we keep "zcl-builtin" stuff, but `test` stuff is not even packaged.

**Review by tecimovic — CHANGES_REQUESTED** [review]

File: src-electron/util/env.js (hunk 1 above)
Comment (brdandu) [reply src-electron/util/env.js]: Done

**Review by tecimovic — APPROVED** [review]

**Comment by brdandu** [comment]
Unify CI failure is a known failure and is being looked into. Will be disabled for the short term.

**Pull request merged**